src/Microsoft.Android.Sdk.ILLink/MarkJavaObjects.cs

```diff
@@ -27,7 +27,14 @@ public override void Initialize (LinkContext context, MarkContext markContext)
 
 		bool IsActiveFor (AssemblyDefinition assembly)
 		{
-			return assembly.MainModule.HasTypeReference ("System.Net.Http.HttpMessageHandler") || assembly.MainModule.HasTypeReference ("Android.Util.IAttributeSet");
+			if (Profile.IsSdkAssembly (assembly))
+				return false;
+			if (Profile.IsProductAssembly (assembly))
+				return false;
+
+			return assembly.MainModule.HasTypeReference ("System.Net.Http.HttpMessageHandler") ||
+				assembly.MainModule.HasTypeReference ("Java.Lang.Object") ||
+				assembly.MainModule.HasTypeReference ("Android.Util.IAttributeSet");
```

Comment:
Checking `IsSdkAssembly()`, `IsProductAssembly()`, and `Java.Lang.Object` is what the `FixAbstractMethodsStep` is already doing, so it seemed appropriate here.